std/haxe/macro/Expr.hx

```diff
@@ -539,6 +548,9 @@ enum ComplexType {
 
 /**
 	Represents a type path in the AST.
+
+	Haxe type paths can only contain underscores `_` or lower-case
+	characters `a-z`.
```

Comment:
That's not what TypePath is as it contains the type name and also type parameters, which makes this information untrue.